sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/query/executor/QueryExecutor.java

```diff
@@ -0,0 +1,619 @@
+@ApplicationScoped
+public class QueryExecutor {
+
+  public static final Accumulator TERM = new Accumulator();
+
+  private final DocumentProperties documentProperties;
+
+  private final StargateRequestInfo requestInfo;
+
+  private final Comparator<DocumentProperty> comparator;
+
+  @Inject
+  public QueryExecutor(DocumentProperties documentProperties, StargateRequestInfo requestInfo) {
+    this.documentProperties = documentProperties;
+    this.requestInfo = requestInfo;
+
+    // create reusable comparator
+    this.comparator = rowComparator();
+  }
+
+  /**
+   * Runs the provided query, then groups the rows into {@link RawDocument} objects with key depth
+   * of 1.
+   *
+   * @see #queryDocs(int, List, int, boolean, ByteBuffer, boolean, ExecutionContext)
+   */
+  public Multi<RawDocument> queryDocs(
+      QueryOuterClass.Query query,
+      int pageSize,
+      boolean exponentPageSize,
+      ByteBuffer pagingState,
+      boolean fetchRowPaging,
+      ExecutionContext context) {
+    return queryDocs(1, query, pageSize, exponentPageSize, pagingState, fetchRowPaging, context);
+  }
+
+  /**
+   * Runs the provided query, then groups the rows into {@link RawDocument} objects according to
+   * {@code keyDepth} primary key values.
+   *
+   * @see #queryDocs(int, List, int, boolean, ByteBuffer, boolean, ExecutionContext)
+   */
+  public Multi<RawDocument> queryDocs(
+      int keyDepth,
+      QueryOuterClass.Query query,
+      int pageSize,
+      boolean exponentPageSize,
+      ByteBuffer pagingState,
+      boolean fetchRowPaging,
+      ExecutionContext context) {
+    return queryDocs(
+        keyDepth,
+        ImmutableList.of(query),
+        pageSize,
+        exponentPageSize,
+        pagingState,
+        fetchRowPaging,
+        context);
+  }
+
+  /**
+   * Runs the provided queries in parallel and merges their result sets according to the selected
+   * primary key columns, then groups the merged rows into {@link RawDocument} objects with key
+   * depth of 1.
+   *
+   * @see #queryDocs(int, List, int, boolean, ByteBuffer, boolean, ExecutionContext)
+   */
+  public Multi<RawDocument> queryDocs(
+      List<QueryOuterClass.Query> queries,
+      int pageSize,
+      boolean exponentPageSize,
+      ByteBuffer pagingState,
+      boolean fetchRowPaging,
+      ExecutionContext context) {
+    return queryDocs(1, queries, pageSize, exponentPageSize, pagingState, fetchRowPaging, context);
+  }
+
+  /**
+   * Runs the provided queries in parallel and merges their result sets according to the selected
+   * primary key columns, then groups the merged rows into {@link RawDocument} objects according to
+   * {@code keyDepth} primary key values.
+   *
+   * <p>Note: all queries must select the same set of columns from the same table.
+   *
+   * <p>Note: queries may use different {@code WHERE} conditions.
+   *
+   * <p>Note: the queries must select all the partition key columns and zero or more clustering key
```

Comment:
Isn't there only one partition key column?

Reply:
This was a leftover comment from V1, where query executor was created with a more generic usage in mind, will adapt.